Spigot-Server-Patches/0325-World-EntityHuman-Lookup-Optimizations.patch

```diff
@@ -0,0 +1,96 @@
+From 49df93e26c208bed8697adcf13a14d1654f96ed5 Mon Sep 17 00:00:00 2001
+From: willies952002 <admin@domnian.com>
+Date: Mon, 30 Jul 2018 02:42:49 -0400
+Subject: [PATCH] World EntityHuman Lookup Optimizations
+
+
+diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
+index ece187129..3625109e9 100644
+--- a/src/main/java/net/minecraft/server/World.java
++++ b/src/main/java/net/minecraft/server/World.java
+@@ -79,6 +79,10 @@ public abstract class World implements GeneratorAccess, IIBlockAccess, AutoClose
+     private final List<TileEntity> c = Lists.newArrayList();
+     private final Set<TileEntity> tileEntityListUnload = Sets.newHashSet(); // Paper
+     public final List<EntityHuman> players = Lists.newArrayList();
++    // Paper start - World EntityHuman Lookup Optimizations
++    public final HashMap<UUID, Entity> entitiesByUUID = com.google.common.collect.Maps.newHashMap();
++    public final HashMap<String, Entity> entitiesByName = com.google.common.collect.Maps.newHashMap();
```

Comment:
Can you change the generic Entity to EntityHuman and add instanceof checks in WorldServer.java instead?

By the way, since you do not remove entries from these two maps, I fear the entity instances referenced in these two maps would bring unnecessary memory overhead.

Reply:
Need to remove entitiesByUUID, it already exists on WorldServer and it should be used.

Move entitiesByName to playersByName instead, so all collections are together.
And manage them together, so the methd in WorldServer that adds them to the byUUID map should manage the player map

Reply:
Need to remove entitiesByUUID, it already exists on WorldServer and it should be used.

Move entitiesByName to playersByName instead, so all collections are together.
And manage them together, so the method in WorldServer that adds them to the byUUID map should manage the player map